### PR TITLE
add invariant check

### PR DESCRIPTION
**Pull request by TyOverby**

### CODE REVIEW

**Comment by TyOverby** [comment]
@gafter: this is the check that I was talking about, does this show that invariants in the parser are being violated or is this expected behavior?

**Comment by TyOverby** [comment]
cc @dotnet/roslyn-compiler @jaredpar

**Comment by TyOverby** [comment]
@dotnet/roslyn-compiler 

Can someone familiar with syntax node invariants take a look at this?  I think that this PR exposes a bug in the parser, but it could be an edge case that is expected.

File: src/Compilers/Core/Portable/Syntax/GreenNode.cs [hunk 1]
```diff
@@ -396,15 +396,21 @@ public bool HasLeadingTrivia
         {
             get
             {
-                return this.GetLeadingTriviaWidth() != 0;
+                var result = this.GetLeadingTriviaWidth() != 0;
+                Debug.Assert((result && GetLeadingTriviaCore() != null) ||
```
Comment (jcouv) [inline src/Compilers/Core/Portable/Syntax/GreenNode.cs]: Maybe use `result == (GetLeadingTriviaCore() != null)` instead?

**Pull request closed without merging**